protocol/bc/byte32.go

```diff
@@ -0,0 +1,89 @@
+package bc
+
+import (
+	"bytes"
+	"database/sql/driver"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"chain/errors"
+)
+
+type Byte32 [32]byte
```

Comment:
Perfect. I had thought that `Byte32` would contain functionality common to both `Hash` and `AssetID`, but on the `entry-protos4` branch I wrote `AssetID` in terms of `Hash` quite naturally, so there's no need for a separate type to factor anything out to.